docs/configuration-lifecycle.md

```diff
@@ -0,0 +1,70 @@
+# Configuration Lifecycle
+
+This document explains how configuration is managed throughout its lifecycle in the Eppo SDK.
+
+## Components Overview
+
+The SDK's configuration management is built around several key components that work together:
+
+- **ConfigurationFeed**: A broadcast channel that serves as the central communication point between components
+- **ConfigurationStore**: Maintains the currently active configuration used for all evaluations
+- **ConfigurationPoller**: Periodically fetches new configurations from the Eppo API
+- **PersistentConfigurationCache**: Persists configuration between application restarts
+
+## Communication Flow
+
+The ConfigurationFeed acts as a central hub through which different components communicate:
+
+![](./configuration-lifecycle.excalidraw.png)
+
+When a new configuration is received (either from network or cache), it's broadcast through the ConfigurationFeed. Components subscribe to this feed to react to configuration changes. Importantly, configurations broadcast on the ConfigurationFeed are not necessarily activated - they may never be activated at all, as they represent only the latest discovered configurations. For components interested in the currently active configuration, the ConfigurationStore provides its own broadcast channel that only emits when configurations become active.
```

Comment:
This is a really exciting new interface - something I want to support in the near future is a concept that the requested configuration is not going to be "complete": we might want to deliver updates via a websocket or the edge JSON will be paged. Either way having a separation between the network response being the canonical source of truth and the configuration store is very important.

Do you feel like the interface you are creating makes us read to push updated in this way? I am wondering if the broadcaster should have a more verbose API that tells listeners whether flags are being added or removed.